Source/HelixToolkit.Shared/Geometry/MeshBuilder.cs

```diff
@@ -3021,8 +3021,10 @@ public void AddTorus(double torusDiameter, double tubeDiameter, int thetaDiv = 3
                 this.AddSphere(new Point3D(), tubeDiameter, thetaDiv, phiDiv);
             }
             // If the second Diameter is zero, we can't build out torus
-            else if (tubeDiameter == 0.0)
-                throw new HelixToolkitException("Torus must have a Diameter bigger than 0");
+            else if (tubeDiameter <= 0.0)
+            {
+                // do nothing, because a torus with tube diameter 0 or below is not visible.
```

Comment:
I think keeping the exception is ok. It lets user know the parameter is invalid.

Reply:
I don't think so. Throwing the exception is unexpected for the user because you can set the `ThetaDiv` and the `PhiDiv` parameter to zero. Both values are also invalid, but there is no exception thrown. The Torus just disapears. I think the behaviour would be more consistent if the TubeDiameter parameter will behave in an equal way. 

In my opinion "do not show the torus" should be enough to let the user know that a parameter is wrong. 

I think this exception was the basic reason for @ikonuk to open the issue #1419.

Reply:
Well, I think all invalid parameters should throw invalid parameter exception. We may want to add the parameter check for all parameters.

Reply:
OK, if all invalid parameters will throw an exception, then the behaviour will be consistent. That is an acceptable solution. I will withdraw my pull request.